### PR TITLE
Docs(UC): Update use cases for Alloc and Dealloc

### LINKED ISSUE #353: Edit use cases for allocation and deallocation in DG

### PR DESCRIPTION
**Pull request by colintkn**
### What this does
Closes #353
Ensures that resident related use cases are updated. 

### How to test
cd to docs/ folder of the project.
Run bundle exec jekyll serve.
Open the jekyll-hosted website in your browser (default address: http://127.0.0.1:4000).
Check that the DG has the changes listed above under the UC-19 and UC-20 section

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -896,41 +893,50 @@ Use case ends.
...
 
 **Extensions**
 
-* 2a. There are no unassigned residents or unassigned.
+* 2a. There are no unassigned residents.
     
     Use case ends.
-* 3a. The given index is invalid.
-    * 3a1. SunRez shows an error message.
+* 3a. The given resident or room do not exist.
```
Comment (stein414) [inline docs/DeveloperGuide.md:906]: do -> does?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -896,41 +893,50 @@ Use case ends.
...
 ### UC-020 Deallocate a Room
 
 **MSS**
 
-1. User requests to list room allocations.
-2. SunRez displays rooms, each with its corresponding allocated resident.
-3. User requests to deallocate a room based on the index from the list displayed in step 2.
-4. SunRez deallocates the room and saves the changes.
+1. User requests to list the residents.
+2. SunRez displays residents, each with its corresponding allocated room.
```
Comment (stein414) [inline docs/DeveloperGuide.md:922]: Maybe change to "each with its corresponding allocated room, if any"?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -896,41 +893,50 @@ Use case ends.
...
 
 **Extensions**
 
-* 2a. There are no unassigned residents or unassigned.
+* 2a. There are no assigned residents.
 
-    Use case ends.
-* 3a. The given index is invalid.
+  Use case ends.
+* 3a. The given resident is does not exist.
```
Comment (stein414) [inline docs/DeveloperGuide.md:931]: remove "is"

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -896,41 +893,50 @@ Use case ends.
...
 
 **MSS**
 
-1. User requests to list room allocations.
-2. SunRez displays rooms, each with its corresponding allocated resident.
-3. User requests to deallocate a room based on the index from the list displayed in step 2.
-4. SunRez deallocates the room and saves the changes.
+1. User requests to list the residents.
+2. SunRez displays residents, each with its corresponding allocated room, if any.
+3. User requests to deallocate a resident from a room based on the index from the list displayed in step 2.
```
Comment (benedictkhoomw) [inline docs/DeveloperGuide.md:923]: Hmm, this seems like it has too much detail about the implementation.

Suggestion: remove the stuff about index and list displayed. Keep it high level like UC-019.

**Review by benedictkhoomw — COMMENTED** [review]
Looks good overall. Thanks for updating.

Left one suggestion for your consideration.

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (colintkn) [reply docs/DeveloperGuide.md:923]: Good point. Thanks.

**Review by benedictkhoomw — APPROVED** [review]
LGTM.

**Pull request merged**